packages/enclave-contracts/contracts/slashing/SlashingManager.sol

```diff
@@ -0,0 +1,370 @@
+
+        if (policy.requiresProof) {
+            // Appeal window is 0 by policy validation, so we dont check for appeal gating
+            require(p.proofVerified, InvalidProof());
+        } else {
+            // Evidence mode with appeals
+            require(block.timestamp >= p.executableAt, AppealWindowActive());
+            if (p.appealed) {
+                require(p.resolved, AppealPending());
+                require(!p.appealUpheld, AppealUpheld()); // approved = appeal upheld => cancel slash, return?
+            }
+        }
+
+        if (p.ticketAmount > 0) {
+            bondingRegistry.slashTicketBalance(
+                p.operator,
+                p.ticketAmount,
+                p.reason
+            );
+        }
+
+        if (p.licenseAmount > 0) {
+            bondingRegistry.slashLicenseBond(
+                p.operator,
+                p.licenseAmount,
+                p.reason
+            );
+        }
+
+        if (policy.banNode) {
+            banned[p.operator] = true;
+            emit NodeBanned(p.operator, p.reason, address(this));
+        }
+
+        emit SlashExecuted(
+            proposalId,
+            p.operator,
+            p.reason,
+            p.ticketAmount,
+            p.licenseAmount,
+            p.executed
+        );
+    }
+
+    // ======================
+    // Appeal Functions
+    // ======================
+
+    /// @inheritdoc ISlashingManager
+    function fileAppeal(uint256 proposalId, string calldata evidence) external {
+        require(proposalId < totalProposals, InvalidProposal());
+        SlashProposal storage p = _proposals[proposalId];
+
+        // Only the accused can appeal
+        require(msg.sender == p.operator, Unauthorized());
+        // Only in the window
+        require(block.timestamp < p.executableAt, AppealWindowExpired());
+        // Only once
+        require(!p.appealed, AlreadyAppealed());
+
+        p.appealed = true;
+
+        emit AppealFiled(proposalId, p.operator, p.reason, evidence);
+    }
+
+    /// @inheritdoc ISlashingManager
+    function resolveAppeal(
+        uint256 proposalId,
+        bool appealUpheld,
+        string calldata resolution
+    ) external onlyGovernance {
+        require(proposalId < totalProposals, InvalidProposal());
+        SlashProposal storage p = _proposals[proposalId];
+
+        require(p.appealed, InvalidProposal());
+        require(!p.resolved, AlreadyResolved());
+
+        p.resolved = true;
+        p.appealUpheld = appealUpheld; // true => cancel slash, false => slash stands
+
+        emit AppealResolved(
+            proposalId,
+            p.operator,
+            appealUpheld,
+            msg.sender,
+            resolution
+        );
+    }
+
+    // ======================
+    // Ban Management
+    // ======================
+
+    /// @inheritdoc ISlashingManager
+    function banNode(address node, bytes32 reason) external onlyGovernance {
```

Comment:
what are the scenarios where we'd ban a node without slashing them first? maybe keeping a list of permabanned addresses (sanctioned ones for instance) that we'd never want to allow to be ciphernodex?